File: packages/cortex-js/src/resources/usage.ts
import type { CortexClient } from '../client.js';
import type { UsageStats, UsageLimits, UsageGetParams, RequestOptions } from '../types.js';

export class Usage {
  private client: CortexClient;

  constructor(client: CortexClient) {
    this.client = client;
  }

  /**
   * Get usage statistics.
   */
  async get(params?: UsageGetParams, options?: RequestOptions): Promise<UsageStats> {
    const url = `${this.client._adminBaseUrl}/api/usage`;
    return this.client._request<UsageStats>('GET', url, {
      query: params as Record<string, string | undefined>,
      requestOptions: options,
    });
  }

  /**
   * Get current rate and usage limits.
   */
  async limits(options?: RequestOptions): Promise<UsageLimits> {
    const url = `${this.client._adminBaseUrl}/api/usage/limits`;
    return this.client._request<UsageLimits>('GET', url, {
      requestOptions: options,
    });
  }
}

File: packages/cortex-dart/lib/src/resources/usage.dart
/// Usage statistics and limits resource.
library;

import '../http_client.dart';
import '../types.dart';

/// Provides access to usage statistics and rate limit endpoints.
class UsageResource {
  final CortexHttpClient _client;
  final String _baseUrl;

  /// Creates a [UsageResource].
  UsageResource({
    required CortexHttpClient client,
    required String baseUrl,
  })  : _client = client,
        _baseUrl = baseUrl;

  /// Gets usage statistics.
  ///
  /// [startDate] and [endDate] optionally filter by date range.
  Future<UsageStats> getStats({
    String? startDate,
    String? endDate,
  }) async {
    final params = <String, String>{};
    if (startDate != null) params['start_date'] = startDate;
    if (endDate != null) params['end_date'] = endDate;

    final json = await _client.get(
      '$_baseUrl/api/usage',
      queryParameters: params.isNotEmpty ? params : null,
    );
    return UsageStats.fromJson(json);
  }

  /// Gets rate and usage limits.
  Future<UsageLimits> getLimits() async {
    final json = await _client.get('$_baseUrl/api/usage/limits');
    return UsageLimits.fromJson(json);
  }
}
